### PR TITLE
Message receipt handler

### PR DESCRIPTION
**Pull request by millibeckers**
Implemented message receipt handler, which only handles waypoint messages as of now

Uses mutation to alter the turkey's waypoint.

Also added racket autosave files to gitignore

### CODE REVIEW

File: gobbler-server.rkt [hunk 1]
```diff
@@ -69,7 +69,7 @@ waiting list.
 ;; An Outgoing is a (make-mail iworld? Server2ClientMessage)
 ;; Represents a message that is to be sent to the iworld
 
-(struct turkey [loc food-eaten waypoint] #:transparent)
+(struct turkey [loc food-eaten [waypoint #:mutable]] #:transparent)
```
Comment (tkindy) [inline gobbler-server.rkt]: The entire turkey struct should probably be mutable. We're going to need to update its location and number of foods eaten in other functions. Might as well do it all in one fell swoop.

File: gobbler-server.rkt [hunk 2]
```diff
@@ -232,17 +235,61 @@ waiting list.
...
 ;; GobblerUniverse iworld? sexp? -> GobblerBundle
 ;; Update the waypoint of the player
-(define (update-waypoint uni world sexp)
-  uni)
+(define (update-waypoint! uni world sexp)
+  (cond [(playing? uni)
+         (cond [(waypoint-message? sexp)
+                (handle-waypoint! uni world sexp)]
+               [else uni])]
+        [else uni]))
```
Comment (tkindy) [inline gobbler-server.rkt]: This function can just be:
```racket
(if (and (playing? uni) (waypoint-message? sexp))
    (handle-waypoint! uni world sexp)
    uni)
```

File: gobbler-server.rkt [hunk 3]
```diff
@@ -232,17 +235,61 @@ waiting list.
...
 
-;; two helper functions that rely on domain knowledge from geometry
+;; Playing iworld? -> [Maybe Player]
+;; Returns the player that corresponds with the given iworld or false if no such
+;; player exists
+(define (get-player playing world)
+  (findf (λ (p) (iworld=? world (player-iworld p))) (ready-players playing)))
+
+;; sexp? -> Boolean
+;; Determines if the given sexp is a valid well-formed message
```
Comment (tkindy) [inline gobbler-server.rkt]: `... well-formed message from the client to the server.` or something like that

File: gobbler-server.rkt [hunk 4]
```diff
@@ -232,17 +235,61 @@ waiting list.
...
+(define (waypoint-message->posn waypoint)
+  (posn (second waypoint) (third waypoint)))
+
+;; Posn -> Boolean
+;; Determines if the waypoint is valid
+(define (waypoint-reachable? posn)
+  (and (>= (posn-x posn) 0)
+       (>= (posn-y posn) 0)
+       (<= (posn-x posn) GAME-SIZE)
+       (<= (posn-y posn) GAME-SIZE)))
```
Comment (tkindy) [inline gobbler-server.rkt]: I would abstract this out to an `in-range?` function that takes the value, the min, and the max.

**Review by tkindy — CHANGES_REQUESTED** [review]

File: gobbler-server.rkt (hunk 4 above)
Comment (millibeckers) [reply gobbler-server.rkt]: Wouldn't that then have to take a max for x and y? thats five things to pass in, I'd rather leave this like it is and then do a posn in range function later if we need it. I can rewrite the comparisons to be cleaner though, I forgot I could do that until just now

Comment (tkindy) [reply gobbler-server.rkt]: I meant that you would call this `in-range?` function for each both the x and y dimensons. i.e.
```racket
(define (in-range? val min max)
  (and (>= val min) (<= val max)))
```

And then this section becomes:
```racket
(and (in-range? (posn-x posn) 0 GAME-SIZE)
     (in-range? (posn-y posn) 0 GAME-SIZE))
```

You could even go one step further and make it `in-game-range?` and hard-code the `0` and `GAME-SIZE` bounds.

Comment (tkindy) [reply gobbler-server.rkt]: To me, that properly abstracts the similarities expressed here.

**Review by tkindy — APPROVED** [review]

**Pull request merged**